components/src/dynamo/sglang/main.py

```diff
@@ -94,6 +94,12 @@ async def init(runtime: DistributedRuntime, config: Config):
     # Readiness gate: requests wait until model is registered
     ready_event = asyncio.Event()
 
+    async def gated_generate(request, context):
+        """Queue requests until model registration completes"""
+        await ready_event.wait()  # Block until model is ready
```

Comment:
is this related to this MR?

@tzulingk  for viz - 

@indrajit96 , @kthui  - is it possible to wait for the ready event before `serve_endpoint` that way the health check will also be in sync - moving to ready only when ready -

Reply:
> is it possible to wait for the ready event before serve_endpoint that way the health check will also be in sync - moving to ready only when ready

If this is not already the case, I think this should be fixed in a separate PR.